split #501

or4nge19 opened the pull request.

or4nge19 committed:
Create Defs.lean

no change from #501

or4nge19 committed:
Create Riemannian.Defs

fixed lints from #501

or4nge19 committed:
Rename PhysLean/Geometry/Metric/Riemannian.Defs to PhysLean/Geometry/Metric/Riemannian/Defs

or4nge19 committed:
Update PhysLean.lean

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+namespace QuadraticForm
+
+variable {K : Type*} [Field K]
+
```
Section heading

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+
+/-- The negative dimension (or index) of a quadratic form is the dimension
+    of a maximal negative definite subspace. -/
+noncomputable def negDim {E : Type*} [AddCommGroup E]
```
Can we make this doc-string more verbose. I can't find a reference for this definition when I search the internet.

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+  exact lt_asymm hq_neg hq_pos
+
+/-- For a positive definite quadratic form, the negative dimension (index) is zero. -/
+theorem rankNeg_eq_zero {E : Type*} [AddCommGroup E]
```
lemma not theorem

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+end QuadraticForm
+
+/-- Helper function to convert the metric tensor at `x` to a quadratic form. -/
+private def pseudoRiemannianMetricValToQuadraticForm
```
Can we change this doc-string to explain what it is, instead of "Helper function".

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+              simp only [LinearMap.mk₂_apply, ContinuousLinearMap.map_add,
+                ContinuousLinearMap.add_apply, symm x]
+              ring⟩
+
```
Section heading

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+continuous bilinear form `gₓ` of constant negative dimension on the tangent space `TₓM`
+at each point `x`. Requires `M` to be `C^{n+1}` smooth.-/
+@[ext]
+structure PseudoRiemannianMetric
```
I think this doc string could be made clearer. Similar to for the charts, e.g. E is usually R etc.

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+lemma toBilinForm_nondegenerate (g : PseudoRiemannianMetric E H M n I) (x : M) :
+    (toBilinForm g x).Nondegenerate := by
+  intro v hv; simp_rw [toBilinForm_apply] at hv; exact g.nondegenerate x v hv
+
```
Section heading. Something like `Flat`

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+
+/-- The "musical" isomorphism (index lowering) from the tangent space to its dual,
+    induced by a pseudo-Riemannian metric. -/
+def flat (g : PseudoRiemannianMetric E H M n I) (x : M) :
```
Can we write the definition of the function in the doc-string

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+
+/-- For a finite-dimensional normed space, the dimension of the continuous dual
+equals the dimension of the original space. -/
+lemma finrank_continuousDual_eq_finrank {𝕜 E : Type*} [NontriviallyNormedField 𝕜]
```
Not used here, and not about anything in this file, so likely is in the wrong place. Same with result above.

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+lemma flatEquiv_apply
+    (g : PseudoRiemannianMetric E H M n I) (x : M) (v w : TangentSpace I x) :
+    (g.flatEquiv x v) w = g.val x v w := rfl
+
```
Section heading

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+
+/-- The "musical" isomorphism (index raising) from the dual of the tangent space to the
+    tangent space, induced by a pseudo-Riemannian metric. This is the inverse of `flatEquiv`. -/
+def sharpEquiv
```
Can we add the definition in the doc-string

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+  rw [g.symm x v (g.sharpL x ω)]
+  rw [← flatL_apply g x (g.sharpL x ω)]
+  rw [flatL_apply_sharpL g x ω]
+
```
Section heading or new file

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+variable [inst_tangent_findim : ∀ (x : M), FiniteDimensional ℝ (TangentSpace I x)]
+
+/-- The value of the induced metric on the cotangent space at point `x`. -/
+noncomputable def cotangentMetricVal (g : PseudoRiemannianMetric E H M n I) (x : M)
```
Can we write defn in doc-string

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+/-- A Riemannian metric of smoothness class `n` on a manifold `M` over `ℝ`.
+    This extends a pseudo-Riemannian metric with the positive definiteness condition. -/
+@[ext]
+structure RiemannianMetric
```
More  verbose doc-string

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+  [TopologicalSpace M] [ChartedSpace H M] [IsManifold I (n + 1) M]
+  [inst_tangent_findim : ∀ (x : M), FiniteDimensional ℝ (TangentSpace I x)] : Type _ where
+  /-- The underlying pseudo-Riemannian metric. -/
+  toPseudoRiemannianMetric : PseudoRiemannianMetric E H M (n) I
```
use `extends PseudoRiemannianMetric E H M n I where` instead of this

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+
+/-- The inverse of the musical isomorphism (index raising), which is an isomorphism
+    in the Riemannian case. This is well-defined because the metric is positive definite. -/
+def sharp
```
Delete, will get for free if use `extend`.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+  g.toPseudoRiemannianMetric.sharpEquiv x
+
+/-- Express a Riemannian metric at a point as a quadratic form. -/
+abbrev toQuadraticForm (g : RiemannianMetric I n M) (x : M) :
```
Delete, will get for free if use `extend`.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+
+/-- The application of a Riemannian metric's quadratic form to a vector. -/
+@[simp]
+lemma toQuadraticForm_apply (g : RiemannianMetric I n M) (x : M)
```
Delete, will get for free if use `extend`.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+  exact g.toQuadraticForm_posDef x
+
+/-- The inner product on the tangent space at point `x` induced by the Riemannian metric `g`. -/
+def inner (g : RiemannianMetric I n M) (x : M) (v w : TangentSpace I x) : ℝ :=
```
Delete, will get for free if use `extend`.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+  g.toPseudoRiemannianMetric.val x v w
+
+@[simp]
+lemma inner_apply (g : RiemannianMetric I n M) (x : M) (v w : TangentSpace I x) :
```
Delete, will get for free if use `extend`.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+variable (g : RiemannianMetric I n M) (x : M)
+
+/-- Pointwise symmetry of the inner product. -/
+lemma inner_symm (v w : TangentSpace I x) :
```
Delete, will get for free if use `extend`.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+
+/-- The inner product space core for the tangent space at a point, derived from the
+Riemannian metric. -/
+noncomputable def tangentInnerCore (g : RiemannianMetric I n M) (x : M) :
```
Delete, will get for free if use `extend`.

or4nge19 replied:
I'm not sure I can do this, the definite condition for the tangentInnerCore requires the metric to be positively defined, which only holds for Riemannian metric, not for PseudoR. 
This is my only reservation, will shortly PR with all other comments addressed (thanks, it all looks nicer now)

jstoobysmith replied:
My bad sorry.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+    linarith [h_inner_zero, h_pos]
+
+/- We Define NormedAddCommGroup and InnerProductSpace structures locally.
+ We DO NOT mark them as instance.-/
```
Add why we don't mark them as instances.

jstoobysmith commented on PhysLean/Geometry/Metric/Riemannian/Defs:
```diff
@@ -0,0 +1,204 @@
+  unfold TangentSpace.metricNormedAddCommGroup
+  simp only [inner_apply]
+  rfl
+
```
Section heading.

jstoobysmith reviewed (commented):
Lots of very small comments. Looks like a lot - but I just went through everything specific instead of making general comments.

jstoobysmith commented on PhysLean/Geometry/Metric/PseudoRiemannian/Defs.lean:
```diff
@@ -0,0 +1,567 @@
+  /-- The metric varies smoothly: Expressed in local coordinates via any chart `e`, the function
+      `y ↦ g_{e.symm y}(mfderiv I I e.symm y v, mfderiv I I e.symm y w)` is `C^n` smooth on the
+      chart's target `e.target` for any constant vectors `v, w` in the model space `E`. -/
+  smooth_in_charts' : ∀ (x₀ : M) (v w : E),
```
This doc string does not match it's definition. You say in the doc-string 'for any chart `e`', but in the definition it is for a fixed chart.

or4nge19 committed:
Update Defs.lean

or4nge19 committed:
Update Defs.lean

or4nge19 committed:
Update Defs.lean

added inner (g : PseudoRiemannianMetric E H M n I) ... as needed for the refactor in Riemannian (using extend)

or4nge19 committed:
Update Defs

or4nge19 committed:
Update Defs

or4nge19 committed:
Update Defs

fix lint

or4nge19 committed:
Update Defs.lean

fix lint

or4nge19 committed:
Update Defs.lean

fix lints...again

or4nge19 committed:
Rename Defs to Defs.lean

or4nge19 committed:
fix

or4nge19 committed:
Update Defs.lean

or4nge19 committed:
fix
